tests/sqlparser_common.rs

```diff
@@ -2103,18 +2108,114 @@ fn parse_select_order_by_nulls_order() {
                 expr: Expr::Identifier(Ident::new("lname")),
                 asc: Some(true),
                 nulls_first: Some(true),
+                with_fill: None,
             },
             OrderByExpr {
                 expr: Expr::Identifier(Ident::new("fname")),
                 asc: Some(false),
                 nulls_first: Some(false),
+                with_fill: None,
             },
         ],
         select.order_by
     );
     assert_eq!(Some(Expr::Value(number("2"))), select.limit);
 }
 
+#[test]
+fn parse_select_order_by_with_fill() {
```

Comment:
Can we add/extend test cases demonstrating behavior for the following cases?

- multiple interpolate clauses e.g `order by a with fill from 10 interpolate (a as b) interpolate (c as d)`
- interpolate inbetween with clauses e.g. `order by a with fill interpolate (a as b), b with fill interpolate (c as d)`
- interpolate without body e.g. `order by a with fill interpolate`
- interpolate with empty body e.g. `order by a with fill interpolate ()`
- interpolate without with fill e.g. `order by a interpolate (a as b)`
- incomplete with fills e.g. `order by a with fill from to 20`, `order by a with fill to step 2` etc

Reply:
> multiple interpolate clauses e.g `order by a with fill from 10 interpolate (a as b) interpolate (c as d)`

This isn't supported syntax. It would be `multiple interpolate clauses e.g order by a with fill from 10 interpolate (a as b, c as d)` ([playground](https://fiddle.clickhouse.com/07af6393-3204-4826-be29-467ab2bcbab4)).

Do you want a test to show this syntax fails?

> interpolate inbetween with clauses e.g. `order by a with fill interpolate (a as b), b with fill interpolate (c as d)`

This syntax isn't supported (the multiple `INTERPOLATE`).

Do you want a test to show this syntax fails?

> interpolate without body e.g. `order by a with fill interpolate`

Done!

> interpolate with empty body e.g. `order by a with fill interpolate ()`

Done!

> interpolate without with fill e.g. `order by a interpolate (a as b)`

As explained https://github.com/sqlparser-rs/sqlparser-rs/pull/1330#discussion_r1666736373, this isn't supported.

Do you want a test for it?

> incomplete with fills e.g. `order by a with fill from to 20, order by a with fill to step 2 etc`

Not entirely sure what this test would look like - each of the options (e.g. `from`, `to`, `step`) require an expression.

Reply:
Ah yeah so for the invalid ones the idea is that they become tests that show the parser fails, heres one for example https://github.com/sqlparser-rs/sqlparser-rs/blob/7022e4227fe00ae4914688ad91e355effd8a159b/tests/sqlparser_clickhouse.rs#L485-L491

Reply:
Sounds good. I'll add a few of those now.

Reply:
Added tests for the failure case in [ebcd31e](https://github.com/sqlparser-rs/sqlparser-rs/pull/1330/commits/ebcd31e6b1ffa7fd282f179f573215036a04b57e):

- ✅ multiple interpolate clauses e.g `order by a with fill from 10 interpolate (a as b) interpolate (c as d)`
- ✅ interpolate inbetween with clauses e.g. `order by a with fill interpolate (a as b), b with fill interpolate (c as d)`
- ✅ interpolate without with fill e.g. `order by a interpolate (a as b)`
- ✅ incomplete with fills e.g. `order by a with fill from to 20, order by a with fill to step 2 etc`